### PR TITLE
PLT-3005 Added confirmation screen to integrations

### PR DESCRIPTION
**Pull request by DavidLu1997**
#### Summary
Adds a confirmation screen showing the relevant details after creating an integration.

This can be utilized as a "View Integration" page in the future. We would need to change:
- Route from `integrations/confirm` to `integrations/view`
- Add integration store listeners for all other integrations (currently only used for OAuth 2.0 Apps)
- Text

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3005

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization files updated

### CODE REVIEW

**Comment by jasonblais** [comment]
+1 other than minor updates:

a) Text tweak to remove link to docs, PR ready here: https://github.com/DavidLu1997/platform/pull/1
b) Remove linespace between `Client ID` and `Client Secret` in the attached image
c) Bold input fields such as `Client ID`, `Client Secret` and `URL(s)` in the attached image

image

**Comment by DavidLu1997** [comment]
@jasonblais changes made

**Comment by jasonblais** [comment]
+1 

Looks great!

**Comment by coreyhulen** [comment]
+1

File: webapp/components/integrations/components/confirm_integration.jsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+        this.state = {
+            type: '',
+            id: '',
+            oauthApps: IntegrationStore.getOAuthApps(userId),
+            loading: !IntegrationStore.hasReceivedOAuthApps(userId)
+        };
+    }
+
+    componentWillMount() {
```
Comment (jwilander) [inline webapp/components/integrations/components/confirm_integration.jsx]: Unless there's a specific reason you used `componentWillMount` then it shouldn't be used. All the state setting here should happen in the constructor and the listeners should be added in `componentDidMount`. `componentWillMount` has been "soft" deprecated and was essentially a way to have constructors for components before ES6

**Comment by DavidLu1997** [comment]
@jwilander `componentWillMount` removed

**Comment by jwilander** [comment]
+1

**Pull request merged**